reactivesocket-client/src/main/java/io/reactivesocket/client/LoadBalancer.java

```diff
@@ -426,37 +502,58 @@ public void onSubscribe(Subscription subscription) {
         }
 
         @Override
-        public void onNext(List<ReactiveSocketFactory<SocketAddress>> newFactories) {
-            List<ReactiveSocketFactory<SocketAddress>> removed = computeRemoved(newFactories);
+        public void onNext(Collection<ReactiveSocketFactory<T>> newFactories) {
             synchronized (LoadBalancer.this) {
```

Comment:
I know you haven't optimized for concurrency, but this seems excessive as for every refresh, this would mean we would not be serving any requests, is my understanding correct?
How would you think it will behave with clusters of thousand servers and a discovery client which isn't great in pushing deltas?